x509-cert/src/certificate.rs

```diff
@@ -73,7 +108,7 @@ impl Default for Version {
 #[cfg_attr(feature = "arbitrary", derive(arbitrary::Arbitrary))]
 #[derive(Clone, Debug, Eq, PartialEq, Sequence, ValueOrd)]
 #[allow(missing_docs)]
-pub struct TbsCertificate {
+pub struct TbsCertificate<PP: ParsingProfile = Rfc5280> {
```

Comment:
I think I like the profile to be attached to all the types better.

Reply:
There are definitely tradeoffs either way.

The main thing that bothers me about it is applying changes to the profile to new fields means you have to add a generic parameter to the type for that field, where having the logic defined on the `Profile` means it can be easily extended and modified without breaking changes.

That said, splitting it up does mean the logic is nicely factored on a type-by-type basis, instead of the profile having all of the responsibility.